pulsar/internal/service_name_resolver.go

```diff
@@ -63,6 +66,8 @@ func (r *pulsarServiceNameResolver) ResolveHost() (*url.URL, error) {
 	if len(r.AddressList) == 1 {
 		return r.AddressList[0], nil
 	}
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
```

Comment:
If we are using a mutex now there is no need for the atomic store. Same for the code below.

Reply:
We should move 
```
r.mutex.Lock()
defer r.mutex.Unlock()
```
To the top since the func `UpdateServiceURL` below can alter the AddressList